### DISCUSSION TITLE
Discrepancy in image quality with and without using entropy coding

### DISCUSSION THREAD

**Discussion by ghost** [InterDigitalInc/CompressAI#120] [Q&A]
Thank you for maintaining these wonderful codes!

When testing pretrained model with my test images, those reconstructed from entropy coded bits(using model.compress() and model.decompress()) gave me around -0.1dB of PSNR compared to the images reconstructed by forwarding them without entropy coding.

However, when I trained the model myself, this discrepancy gets worse, almost up to -2dB.
I trained it until validation mse and bpp converges, and also optimized auxiliary loss

I don't understand where this difference comes from, and hope someone could give me some insight.

**Comment by fracape** [comment]
Hi, 
Sorry for the late reply, can you tell me which model. Are you training using the provided exemplary train.py? A minor difference of bitrate between compress (true inference with entropy coding) and forward is normal since forward only estimates the entropy. However, the quality of the reconstruction (since you are talking in dB) should not differ at the same operating point, except if you adapt/modified something else to remain at the same bpp. (Note that if you compress and decompress on different platforms, depending on the model that you use, you can face big discrepancies because of floating point operations, especially if you use a hyperprior-based model which uses a NN to estimate a probability model. However, in this case, I think you would get even lower PSNR values)
